mail_tracking/models/mail_tracking_email.py

```diff
@@ -473,3 +473,26 @@ def event_process(self, request, post, metadata, event_type=None):
         # - return 'NONE' if this request is not for you
         # - return 'ERROR' if any error
         return "NONE"  # pragma: no cover
+
+    def _get_old_mail_tracking_email_domain(self, max_age_days):
+        target_write_date = fields.Datetime.subtract(
+            fields.Datetime.now(), days=max_age_days
+        )
+        return [("write_date", "<", target_write_date)]
+
+    @api.autovacuum
+    def _gc_mail_tracking_email(self, max_age_days=180, limit=5000):
```

Comment:
`max_age_days` should be configurable through a system parameter as well.
We could replace the current boolean config option by an integer instead, 0 meaning nothing is deleted, or a number of retention days to enable it.